### PR TITLE
[NG] wizard unit tests and deep bug fixing

### PR DESCRIPTION
**Pull request by mathisscott**
• unit tests for the wizard revealed some deep, deep issues with the navigation service, current pages, and ngFor
• thus far, users have not been impacted by these issues but they would have been — we have demos on the website failing due to these problems
• these issues took a lot of time to refactor and prove out via unit tests
• used TestContext helper for all wizard unit tests
• tested all demos on app to make sure they worked as expected

Tested in:
✔︎ Chrome

Signed-off-by: Scott Mathis <smathis@vmware.com>

[NG] unit tests

Signed-off-by: Scott Mathis <smathis@vmware.com>

[NG] unit tests

• moved shared components to a test-components folder

Signed-off-by: Scott Mathis <smathis@vmware.com>

[NG] dox for wizard.ts

Signed-off-by: Scott Mathis <smathis@vmware.com>

[NG] dox for wizard.ts

Signed-off-by: Scott Mathis <smathis@vmware.com>

[NG] wizard unit tests and deep bug fixing

• unit tests for the wizard revealed some deep, deep issues with the navigation service, current pages, and ngFor
• thus far, users have not been impacted by these issues but they would have been — we have demos on the website failing due to these problems
• these issues took a lot of time to refactor and prove out via unit tests
• used TestContext helper for all wizard unit tests

Tested in:
✔︎ Chrome

Signed-off-by: Scott Mathis <smathis@vmware.com>

### CODE REVIEW

File: src/clarity-angular/wizard/wizard-button.ts [hunk 1]
```diff
@@ -107,7 +107,7 @@ export class WizardButton {
         let nav = this.navService;
```
Comment (adityarb88) [inline src/clarity-angular/wizard/wizard-button.ts]: i know this isn't touched in this PR but it looks like `nav` isn't being used anywhere, no?

File: src/clarity-angular/wizard/wizard.html [hunk 2]
```diff
@@ -12,7 +12,7 @@
     [clrModalSkipAnimation]="stopModalAnimations"
     [clrModalGhostPageState]="ghostPageState"
     [clrModalOverrideScrollService]="isStatic"
-    clrModalPreventClose="true"
+    [clrModalPreventClose]="true"
```
Comment (adityarb88) [inline src/clarity-angular/wizard/wizard.html]: so isn't of passing a `string` we are now passing a `boolean`. Has the type changed or was this a bug before?

File: src/clarity-angular/wizard/providers/page-collection.mock.ts [hunk 3]
```diff
@@ -23,7 +23,8 @@ export class PageCollectionMock {
         return this._stepItemIdWasCalled;
     }
 
-    public previousPageIsCompleted(): boolean {
-        return true;
+    public _previousPageIsCompleted = true;
+    public previousPageIsCompleted(page: any = null): boolean {
```
Comment (mathisscott) [inline src/clarity-angular/wizard/providers/page-collection.mock.ts]: PageCollection needs to know which page you are asking about. You can ask it if any page's previous page is completed. So, not really a getter.

File: src/clarity-angular/wizard/wizard-button.ts (hunk 1 above)
Comment (mathisscott) [inline src/clarity-angular/wizard/wizard-button.ts]: I'll take a look and remove it if that is the case.

File: src/clarity-angular/wizard/wizard.html (hunk 2 above)
Comment (mathisscott) [inline src/clarity-angular/wizard/wizard.html]: Bug? What bug?

I never write bugs.

This is obviously some kind of... thing... that... needed to be changed to a boolean?

Comment (adityarb88) [inline src/clarity-angular/wizard/wizard.html]: lol. you are right! the comment that i wrote was a bug. sorry

File: src/clarity-angular/wizard/wizard-button.ts (hunk 1 above)
Comment (mathisscott) [inline src/clarity-angular/wizard/wizard-button.ts]: I've verified `nav` is used on ln 118, 127, and 131.

**Pull request merged**